tests/rados/pool_tests.py

```diff
@@ -565,6 +565,122 @@ def run(ceph_cluster, **kw):
+        log.debug(pool_stat)
+        total_objs = pool_stat["stats"]["objects"]
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=10, timeout=10)
+        pool_stat_post = rados_obj.get_cephdf_stats(pool_name=pool_name)
+        log.debug(pool_stat_post)
+        if pool_stat_post["stats"]["objects"] != total_objs:
+            log.error(
+                "Write ops should not have been possible, number of objects in the pool have changed"
+            )
+            raise Exception(
+                f"Pool {pool_name} has {pool_stat_post['stats']['objects']} objs | Expected {total_objs} objs"
+            )
+        else:
+            log.info(
+                "RADOS PUT op failed as expected since cluster is in degraded state!! "
+            )
+
+        # Setting the min-size as 1. The IOs must be resumed and the PGs should become active+clean state.
+        if not rados_obj.set_pool_property(pool=pool_name, props="min_size", value=1):
+            log.error(f"failed to set min_size 1 on pool {pool_name}")
+            raise Exception("Config set failed on pool")
+
+        min_size_val = rados_obj.get_pool_property(pool=pool_name, props="min_size")
+        log.info(f"min_size of the pool after modification - {min_size_val}")
+
+        pg_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+        log.info(
+            f"Post modification, Pg set fetched from pool id {pool_id} for pg id {pgid} is {pg_set}"
+        )
+        log.info(f"Current OSDs in acting set {pg_set}")
+        osd_tree_op = rados_obj.run_ceph_command(cmd="ceph osd tree")
+        log.info(f"After bringing down osds, output from ceph osd tree : {osd_tree_op}")
+        time.sleep(100)
+
+        # verify IOs are resumed after sometime post setting min_size to 1
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=10, timeout=10)
+        pool_stat_post = rados_obj.get_cephdf_stats(pool_name=pool_name)
+        log.debug(pool_stat_post)
+        expected_total_objs = total_objs + 10
+        if pool_stat_post["stats"]["objects"] <= total_objs:
+            log.error(
+                "Write ops should have been possible, number of objects in the pool have NOT changed"
+            )
+            raise Exception(
+                f"Pool {pool_name} has {pool_stat_post['stats']['objects']} objs | Expected {expected_total_objs} objs"
+            )
+        else:
+            log.info("RADOS PUT op is successful as expected!! ")
+
+        log.info(
+            "With cluster in degraded state and OSDs in acting set equal to min_size, IOs are served"
+        )
+
+        log.info(
+            "Tested the behaviour of cluster in degraded state and impact of reducing min-size"
+        )
+
+        # Checking the PG state. There Should not be inactive state
+        pg_state = rados_obj.get_pg_state(pg_id=pgid)
+        if any("inactive" in key for key in pg_state.split("+")):
```

Comment:
If the pgid is in inactive state, will the IOPs which got completed successfully after min_size change be possible? If not, then this check becomes redundant. Should we consider performing this check before starting the second round of rados put(the one we are expecting to pass)?
Just my hunch, open to discussion

Reply:
The PGs will not go inactive state in degraded state as well. This is just an final check on the cluster before handing over the cluster for further execution of tests so that cluster is healthy for next executions.